packages/story-editor/src/components/checklist/checks/storyAmpValidationErrors.js

```diff
@@ -35,38 +40,22 @@ export async function getStoryAmpValidationErrors({ link, status }) {
   try {
     const response = await fetch(link);
     const storyMarkup = await response.text();
-    const { status: markupStatus, errors } =
-      window.amp.validator.validateString(storyMarkup);
+
+    await window.amp.validator.init();
+
+    /**
+     * @type {ValidationResult}
+     */
+    const validationResult = window.amp.validator.validateString(storyMarkup);
+    const { status: markupStatus, errors } = validationResult;
 
     if ('FAIL' !== markupStatus) {
       return false;
     }
 
     const filteredErrors = errors
       .filter(({ severity }) => severity === 'ERROR')
-      .filter(({ code, params }) => {
-        // Filter out errors that are covered in other checks
-        // Already covered by metadata checks.
-
-        // Missing story poster
-        if ('MISSING_URL' === code && params?.[0].startsWith('poster')) {
-          return false;
-        }
-
-        // Missing publisher logo
-        if (
-          'MISSING_URL' === code &&
-          params?.[0].startsWith('publisher-logo')
-        ) {
-          return false;
-        }
-        // Missing video posters
-        if ('INVALID_URL_PROTOCOL' === code && params?.[0].startsWith('src')) {
-          return false;
-        }
-
-        return true;
-      });
```

Comment:
Why was this contextual handling of error codes removed?

This sounds like any `MISSING_URL` errors are filtered, while we only need to filter them when the error is for specific elements (like story poster or publisher logo).

But if `MISSING_URL` happens on some other element, we still want to surface that error.

So I think this needs to be reverted.

Reply:
Think it's as simple as I misunderstood. I'll pop it back in and swap this PR back to a draft pending what you find out about the wrong error code.